### PR TITLE
Students

### PR DESCRIPTION
**Pull request by svetlana-suslova**
a fix on server needed:

1) routes.ts
 helper.delete('/api/student/delete/:id', studentController.deleteStudent);

2) studentController.ts
async function deleteStudent(req, res) {
  try {
    let data = await helper.loadSchema(req.params, {
      id: Joi.number().required(),
    });
    await studentRepository.deleteStudent(data.id);
  ...
}

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,6 +43,7 @@
     "jquery": "3.5.1",
     "js-cookie": "2.2.1",
     "lodash": "4.17.20",
+    "moment": "2.29.1",
```
Comment (erik-sytnyk) [inline package.json:46]: we use `date-fns` library instead of `moment`

File: src/action_types/actionTypes.ts [hunk 2]
```diff
@@ -1,5 +1,12 @@
 export const ASYNC_ACTION_START = 'common/ASYNC_ACTION_START';
 export const ASYNC_ACTION_END = 'common/ASYNC_ACTION_END';
+export const CONFIRM_ACTION = 'common/CONFIRM_ACTION';
```
Comment (erik-sytnyk) [inline src/action_types/actionTypes.ts:3]: For new structure we do separate action types into different files

Please see new template structure for reference

File: src/actions/studentActions.ts [hunk 3]
```diff
@@ -5,9 +5,52 @@ import service from 'services/studentService';
...
   return helper.dispatchAsyncAction(async (dispatch) => {
     let statistics = await service.getStudentsStatistics();
     dispatch(loadStudentsStatisticsSuccess(statistics));
   }, null);
 };
+
+export const loadStudents = (sortOrder, search, pageNumber, pageSize) => {
+  return helper.dispatchAsyncAction(async (dispatch) => {
+    let students = await service.getStudents(sortOrder, search, pageNumber, pageSize);
+    dispatch(loadStudentsSuccess(students.rows));
```
Comment (erik-sytnyk) [inline src/actions/studentActions.ts:30]: do we need 2 dispatch (s) here? can it be one `loadStudentsSuccess(students)`

**Comment by svetlana-suslova** [comment]
a new pull request with fixes will be created

**Pull request closed without merging**